test/specs/modules/Checkbox/Checkbox-test.js

```diff
@@ -201,6 +201,50 @@ describe('Checkbox', () => {
     })
   })
 
+  describe('click propagation', () => {
+    const assertions = [
+      {
+        description: 'propagates once on "label" without "id"',
+        id: undefined,
+        target: 'label',
+        propagates: true,
+      },
+      {
+        description: 'does not propagate on "input" without "id"',
+        id: undefined,
+        target: 'input',
+        propagates: false,
+      },
+      {
+        description: 'propagates once on "label" with "id"',
+        id: 'foo',
+        target: 'label',
+        propagates: true,
+      },
+      {
+        description: 'does not propagate on "input" with "id"',
+        id: 'foo',
+        target: 'input',
+        propagates: false,
+      },
+    ]
+
+    assertions.forEach((assertion) => {
+      it(assertion.description, () => {
+        const onClick = sandbox.spy()
+        wrapperMount(
+          <div role='presentation' onClick={onClick}>
+            <Checkbox id={assertion.id} />
+          </div>,
+        )
+        const target = document.querySelector(`.ui.checkbox ${assertion.target}`)
+        domEvent.click(target)
+        if (assertion.propagates) onClick.should.have.been.calledOnce()
+        else onClick.should.not.have.been.called()
```

Comment:
```diff
-if (assertion.propagates) onClick.should.have.been.calledOnce()
-else onClick.should.not.have.been.called()
+onClick.should.have.callCount(Number(assertion.propagates))
```

We can test the call count directly, https://github.com/domenic/sinon-chai

Reply:
Nice idea!